pkg/kubelet/volumemanager/reconciler/reconciler.go

```diff
@@ -507,16 +507,6 @@ func (rc *reconciler) reconstructVolume(volume podVolume) (*reconstructedVolume,
 			newMounterErr)
 	}
 
-	// Check existence of mount point for filesystem volume or symbolic link for block volume
```

Comment:
Have some concern about removing this check. In the case of volume dir exists, but is not mounted. During reconstruction, the volume is not in desired state, so updateState will add this volume into actual state. Right after it, pod is added, and desired state has this volume, then reconcile will not trigger mount since both actual state and desired state has this volume?

Reply:
There is really no way how to check if a CSI volume is mounted - it does not need to be mounted at all. And the check is testing a wrong directory anyway.

The only way how to make sure the volume is complete is to call `SetUp` (or `TearDown` if the volume is not needed).

For CSI, we could perhaps squeeze `SetUp()` call directly into CSI volume plugin `ConstructVolumeSpec`, I am not sure it's a good idea though.

I remember some discussions that reconstructed volumes would be marked as "uncertain" until SetUp() confirms they're up to date (or broken). Was it ever implemented?

Reply:
Invoking plugin during reconstruction was problematic too: https://github.com/kubernetes/kubernetes/issues/72500.

I thought we only use reconstruction results to determine if cleanup needs to be done for pods that have been deleted?

Reply:
> I thought we only use reconstruction results to determine if cleanup needs to be done for pods that have been deleted?

If a pod still exists but is being deleted at reconstruction time, we need to add its volumes to ASW so they're cleaned when the pod is finally deleted.